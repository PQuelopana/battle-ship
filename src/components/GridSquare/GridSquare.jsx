import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types';

import './GridSquare.scss'

export const GridSquare = ({
    column,
    row,
    onClick,
    isOccupied,
    color,
    existShot,
    hideShip,
    gameOver,
}) => {
    const [stateClassIsOccupied, setStateClassIsOccupied] = useState([])
    const [stateClassColor, setStateClassColor] = useState([])
    const [stateClassExistShot, setStateClassExistShot] = useState([])
    const [stateClassGameOver, setStateClassGameOver] = useState([])
    const [stateClassTimerIsActive, setStateClassTimerIsActive] = useState([])

    const handleClick = () => {
        if (typeof onClick === 'function') onClick(row, column)
    }

    useEffect(() => {
        setStateClassIsOccupied(isOccupied ? ['is-occupied'] : [])
    }, [isOccupied])

    useEffect(() => {
        setStateClassColor(!hideShip && !existShot && color ? [`is-${ color }`] : [])
    }, [hideShip, color, existShot])

    useEffect(() => {
        setStateClassTimerIsActive(!hideShip ? ['is-timer-active'] : [])
    }, [hideShip])

    useEffect(() => {
        setStateClassExistShot(existShot ? ['exist-shot'] : [])
    }, [existShot])

    useEffect(() => {
        setStateClassGameOver(gameOver ? ['game-over'] : [])
    }, [gameOver])

    return (
        <div
            className={[].concat(['square'], stateClassColor, stateClassIsOccupied, stateClassExistShot, stateClassGameOver, stateClassTimerIsActive).join(' ')}
            onClick={handleClick}
        />
    )
}

GridSquare.propTypes = {
    column: PropTypes.number.isRequired,
    row: PropTypes.number.isRequired,
    // onClick: PropTypes.func.isRequired,
}

GridSquare.defaultProps = {
    color: '',
    isOccupied: false,
    existShot: false,
    hideShip: false,
    gameOver: false,
}
