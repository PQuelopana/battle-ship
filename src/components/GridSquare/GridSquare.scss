@import '../../assets/scss/global';

.square {
    border: fluid(1) solid #7d7d7d;
    transition: all .5s;

    &.exist-shot {
        background-color: orange;
    }

    &.is-occupied {

        &.is-black {
            background-color: black;
        }

        &.is-red {
            background-color: red;
        }

        &.is-blue {
            background-color: blue;
        }

        &.is-yellow {
            background-color: yellow;
        }
    }

    &.game-over, &.is-timer-active {
        opacity: .5;
    }
}