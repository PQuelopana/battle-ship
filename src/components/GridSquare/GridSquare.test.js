// import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import { GridSquare } from './GridSquare'

describe('tests in <GridSquare />', () => {
    test('should render <GridSquare /> correctly', () => {
        const wrapper = shallow(
            <GridSquare
                column={1}
                row={1}
                onClick={() => console.log('onClick')}
            />
        )

        expect(wrapper).toMatchSnapshot()
    })
})
