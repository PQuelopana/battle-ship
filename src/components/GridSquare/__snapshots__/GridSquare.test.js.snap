// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GridSquare /> should render <GridSquare /> correctly 1`] = `
<div
  className="square"
  onClick={[Function]}
/>
`;
